jenkins/run-pytests

```diff
@@ -18,10 +20,15 @@ if [[ ${flake_sts} -ne 0 ]]; then
     printf -- "FAILED: flake8 --verbose .\n" >&2
 fi
 printf -- "\n"
-pytest ./lib/pbench/test/unit/common ./lib/pbench/test/unit/agent ./lib/pbench/test/unit/server ./lib/pbench/test/functional
+pytest --cov --cov-append \
+    --cov-report xml:/tmp/${USER}/cov/cov.xml \
```

Comment:
We have the same inconsistent usage of `/tmp/${USER}` vs `{envdir}`... did you revert that deliberately?

Reply:
I don't think so, but maybe this was a good thing to have made this mistake because it gives me a chance to explain this behavior.

The non-`jenkins-` `pytest` runs are expected to be run by users, so the report generated is an `html` report, intended to be more human readable than the XML.  The Jenkins runs, on the other hand, need the XML to upload for JUnit integration in the Jenkins dashboard, and there is no facility for capturing the `html` directory hierarchy.

So in `tox.ini` `py3-agent` and `py3-server` capture separate coverage for each of them as an `html` report.  The Jenkins job runs both the agent and the server code together, and generates one XML report for uploading to the Jenkins instance.